ComposableArchitecture/ComposableArchitecture/Sources/Pullback.swift

```diff
@@ -0,0 +1,47 @@
+import CasePaths
+import Foundation
+
+/// An optic is "a composable notion of substructure." There are many types of
+/// optics; here we declare the protocols for two common optics that we will
+/// make use of throughout the ComposableArchitecture. A Lens describes how to
+/// transform a superstructure into its component parts (and back again). A
+/// Prism distinguishes a particular case from its variants (and back again).
+/// https://hackage.haskell.org/package/optics-0.1/docs/Optics.html
+public enum Optics {
```

Comment:
Lets discuss this before merging. I would love to understand these with examples.